### PR TITLE
Remove timer overflow underflow

### PR DESCRIPTION
**Pull request by vicsn**
When proving large batches of circuits, the profiler messages still crashed.

I see the `cargo fmt` and `Cargo.lock` cleanup may conflict with the other open PR. Feel free to just use the single commit afc1183cd02f58ed4de8fa4910260e7e82e374a4 or ask me to rebase if the other PR is approved.

### CODE REVIEW

File: profiler/src/lib.rs [hunk 1]
```diff
@@ -40,15 +40,12 @@ pub mod inner {
 
             let msg = $msg();
             let start_info = "Start:".yellow().bold();
-            let indent_amount = 2 * NUM_INDENT.fetch_add(0, Ordering::Relaxed);
+            let indent_amount = NUM_INDENT.fetch_add(0, Ordering::Relaxed).saturating_mul(2usize);
```
Comment (ljedrz) [inline profiler/src/lib.rs:43]:
```suggestion
            let indent_amount = NUM_INDENT.load(Ordering::Relaxed).saturating_mul(2usize);
```
same here

Comment (vicsn) [reply profiler/src/lib.rs]: Agreed, pushed and squashed

File: profiler/src/lib.rs [hunk 2]
```diff
@@ -82,8 +79,10 @@ pub mod inner {
             let end_info = "End:".green().bold();
             let message = format!("{} {}", $time.msg, $msg());
 
-            NUM_INDENT.fetch_sub(1, Ordering::Relaxed);
-            let indent_amount = 2 * NUM_INDENT.fetch_add(0, Ordering::Relaxed);
+            NUM_INDENT
+                .fetch_update(Ordering::Relaxed, Ordering::Relaxed, |x| if x > 0 { Some(x - 1) } else { Some(x) })
+                .unwrap();
+            let indent_amount = NUM_INDENT.fetch_add(0, Ordering::Relaxed).saturating_mul(2usize);
```
[the remark this answers is not in the document]
Comment (vicsn) [reply profiler/src/lib.rs]: Agreed, pushed and squashed

**Comment by vicsn** [comment]
Closing this for now, I see that our tests have an uneven number of start_timer! and end_timers!, so perhaps it is good that we're failing loudly... will streamline that and adjust this PR so we throw a clearer error

**Pull request closed without merging**